code/datums/components/storage/storage.dm

```diff
@@ -551,6 +567,9 @@
 //This proc return 1 if the item can be picked up and 0 if it can't.
 //Set the stop_messages to stop it from printing messages
 /datum/component/storage/proc/can_be_inserted(obj/item/I, stop_messages = FALSE, mob/M)
+	can_hold_full = typecacheof(can_hold)
+	cant_hold_full = typecacheof(cant_hold)

```

Comment:
Making these lists every time can_be_inserted is called is expensive. Saving the lists so you don't need to make them is also expensive. You need to cache unique lists so that at the very least you don't have duplicate saved lists